openverse_catalog/dags/providers/provider_api_scripts/rawpixel.py

```diff
@@ -1,193 +1,282 @@
+    def _get_category(metadata: dict) -> str | None:
+        keywords = set(metadata.get("popular_keywords", []))
+        if "public domain art" in keywords:
+            return prov.ImageCategory.DIGITIZED_ARTWORK.value
+        if "image" in keywords or "photo" in keywords:
+            return prov.ImageCategory.PHOTOGRAPH.value
+        if "clipart" in keywords:
+            return prov.ImageCategory.ILLUSTRATION.value
         return None
-    img_url, width, height = _get_image_properties(image, foreign_url)
-    if not img_url:
-        return None
-    title, owner = _get_title_owner(image)
-    meta_data = _get_meta_data(image)
-    tags = _get_tags(image)
-
-    # TODO:How to get license_url, creator_url, source, watermarked?
-    license_info = get_license_info(
-        license_=license_,
-        license_version=version,
-    )
-    return image_store.add_item(
-        foreign_landing_url=foreign_url,
-        image_url=img_url,
-        license_info=license_info,
-        foreign_identifier=str(foreign_id),
-        width=str(width) if width else None,
-        height=str(height) if height else None,
-        title=title if title else None,
-        meta_data=meta_data,
-        raw_tags=tags,
-        creator=owner,
-    )
-
-
-def _process_pages(total, result, page):
-    img_ctr = 0
-    is_valid = True
-
-    if not total:
-        total = 0
-        is_valid = False
 
-    while (img_ctr < total) and is_valid:
-        logger.info(f"Processing page: {page}")
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        # verify the license and extract the metadata
+        if not (foreign_id := data.get("id")):
+            return None
 
-        for img in result:
-            total_images = _process_image_data(img)
-            img_ctr = total_images if total_images else img_ctr
+        if not (foreign_url := data.get("url")):
+            return None
 
-        page += 1
-        total, result = _get_image_list(page)
+        if not (metadata := data.get("metadata")):
+            return None
 
-        if not result:
-            is_valid = False
+        license_info = get_license_info(metadata["licenseUrl"])
+        if license_info == NO_LICENSE_FOUND:
+            return None
 
-        if not total:
-            total = 0
-            is_valid = False
+        if not (image_url := self._get_image_url(data, self.full_size_option)):
+            return None
 
-    return img_ctr
+        width, height = self._get_image_properties(data)
+        return {
+            "foreign_landing_url": foreign_url,
+            "image_url": image_url,
+            "license_info": license_info,
+            "foreign_identifier": foreign_id,
+            "width": width,
+            "height": height,
+            "title": self._get_title(metadata),
+            "meta_data": self._get_meta_data(data, metadata),
+            "raw_tags": self._get_tags(metadata),
+            "creator": self._get_source(data),
```

Comment:
Should this method be named `_get_creator`? It was a bit confusing since we have a `source` column.

Reply:
Oh great call!